src/api/cat.js

```diff
@@ -1,7 +1,32 @@
 'use strict'
 
-const argCommand = require('../cmd-helpers').argCommand
+const bs58 = require('bs58')
+const isIPFS = require('is-ipfs')
+const promisify = require('promisify-es6')
 
 module.exports = (send) => {
-  return argCommand(send, 'cat')
+  const cat = promisify((multihash, callback) => {
+    try {
+      multihash = cleanMultihash(multihash)
+    } catch (err) {
+      return callback(err)
+    }
+    send('cat', multihash, null, null, function (err, result) {
+      if (err) {
+        return callback(err)
+      }
+      return callback(null, result)
+    })
+  })
+  return cat
+}
+
+function cleanMultihash (multihash) {
+  if (!isIPFS.multihash(multihash)) {
+    throw new Error('not valid multihash')
+  }
+  if (Buffer.isBuffer(multihash)) {
```

Comment:
I've updated the <a href="https://github.com/ipfs/interface-ipfs-core/pull/21/files#diff-04c6e90faac2675aa89e2176d2eec7d8R70">readme</a> in interface-core, hopefully that simplifies the statement. A buffer is just a node.js default utf-8 buffer of a multihash, and a ```Qm...``` string is base58 encoded.